manifests/init.pp

```diff
@@ -37,7 +37,7 @@
   String $omapi_algorithm                                 = 'HMAC-MD5',
   Optional[String] $omapi_key                             = undef,
   Boolean $authoritative                                  = true,
-  $extra_config                                           = '',
+  Optional[Variant[Array,String]] $extra_config           = undef,
```

Comment:
How about defaulting to ```[]``` and make it non-optional?

Reply:
Why should it be non-optional?

Reply:
Easier typing? An empty array as default means it's also optional for users but you could drop some code.

Reply:
ah now I got your point. I will update it and remove the optional